include/deal.II/lac/petsc_precondition.h

```diff
@@ -957,6 +957,131 @@ namespace PETScWrappers
     AdditionalData additional_data;
   };
 
+  /**
+   * A class that implements the interface to use the BDDC preconditioner from
+   * PETSc (<a
+   * href="https://petsc.org/release/docs/manualpages/PC/PCBDDC.html">PCBDDC</a>),
+   * which is a two-level, substructuring, non-overlapping domain decomposition
+   * preconditioner. Details of the implementation can be found in "S Zampini,
+   * SISC (2016)". It mainly consists of two elements:
+   *
+   * <ul>
+   *   <li> Local solvers: Solvers for each subdomain. These are performed concurrently by each processor
+   *   <li> A coarse solver: Continuity between each subdomain is imposed in a small number of DoFs, referred to as <em>primal DoFs</em>. This solver solves such problem.
+   * </ul>
+   *
+   * The size of the primal space is determined through the @p AdditionalData parameters.
+   *
+   * @ingroup PETScWrappers
+   */
+  class PreconditionBDDC : public PreconditionBase
+  {
+  public:
+    /**
+     * Standardized data struct to pipe additional flags to the
+     * preconditioner.
+     */
+    struct AdditionalData
+    {
+      /**
+       * Constructor. Note that BDDC offers a lot more options to set
+       * than what is exposed here.
+       */
+      AdditionalData(const bool                    use_vertices = true,
+                     const bool                    use_edges    = false,
+                     const bool                    use_faces    = false,
+                     const bool                    symmetric    = false,
+                     const unsigned int            coords_cdim  = 0,
+                     const types::global_dof_index coords_n     = 0,
+                     const PetscReal *             coords_data  = nullptr);
```

Comment:
This was mentioned in at least one other place, but to match the convention used elsewhere in the library this really should be a `std::vector<Point<spacedim>>` vector. Avoiding templates is good in general though this (like the cgal additional data struct) is a place where we should use them.

Reply:
I understand, I am working on this now. 

An extreme side note, but I have taken long to start because my reinstallation has been a bit difficult due to the migration of METIS to github only. I have some workarounds for candi, if you every face the problem.

Reply:
I usually just let PETSc download and install METIS for me to avoid these problems.